### ISSUE TITLE
Question: Is this only for fast audio etc or can it be used for effects like this?

### ISSUE THREAD

**Issue by GamingDaveUk** [ryanontheinside/ComfyUI_RyanOnTheInside#23]
I originally asked this on one of your youtube video's but I felt off about sharing a link to what I want to achieve as that felt a bit spammy on youtube. But this is the sort of effect I am after:
https://www.youtube.com/watch?v=SK_uKQIswsA

This was created with vizzy, and its the following:
A video on a loop in the back ground (I created the video in comfyui but more after if I can use a created video, if i can then i can create a workflow to create nearly anything i need lol)
A spectrogram that is mirrored (top to bottom) this works well for speech and is where i suspect your back would come in well.
A mp3 pre-created. 

Vizzy will put these together though it takes time. But i would prefer to use comfyui, for one I dont just create the spoken word videos but ai music too, these are mostly for mates, comfyui node based approach would allow for more versitility, and for the logs, batch processing would be handy. 

So is it possible with your node suite to do the above?
ie loop a video in the background, create a spectrogram of the voice in the forground and combine it all into a video?

**Comment by GamingDaveUk** [comment]
i had some time today to play around with it.
First I tried the smoke workflow, however that one relies on specific video's, interesting workflow but since its sd1.5 I am guessing its a very old one. The result is very specific thanks to the mask and video being used but is very cool too...not the effect I am going for, i think we have very different music tastes lol.

I then decided to try the basics to understand the node set. I took a look at your audio seperation video... I thought that would be the jackpot as I essentially want to use the vocals for the visualisation (spectrograth) however it needed a video to drive the length. I loaded in an audio instead of a video then loaded in a still image as a background image (you stated in the video a blank frame can be used, so this seemed perfect) hooked up the feature line to the vocals. however the result was a still image in video form.

I attempted to loop the end video and got a still image that stayed on the screen for a second or two with the start of the audio but no animation to it... infact the image showing the flow in your video is showing no flow in mine. I Have run out of time until friday now. My gut is telling me the suite does not have what i need it to do (not saying it like a bad thing, we code to create things we wish to see and use, thats what makes it a hobby. despite this seeming to be not what i need i am impressed with what you can create with it in your videos, the audio seperation may come in handy for me, and if i can ever find a node that will do what i need then the nodes you have will only help there.... my point is that i dont mean this in the negative way that text may make it sound.) 
If I am wrong in my assumption and your node suite can be used to create a video like I shared (though spoken logs is not all I wish to use it for, infact i tested it with a song I created with AI for a game me and a friend are playing, not a spoken log) then please let me know how even if you just tell me what nodes to look at I will have another look friday.

**Comment by GamingDaveUk** [comment]
Well first hurdle jumped when I found out that filenames is an array that does include the filename... yes the name suggests that but it wasnt a string etc so wasnt sure what to do with it... a specific node helped there:
![image](https://github.com/user-attachments/assets/53a0683f-1ae2-406c-9073-75b2d4ce8cd8)
I have the loaded video feed its frames into "destination" for image composite masked... but really no clue if that bit is right* (it uses all 64gb of my ram when it hits that spot and my pc does its best to act lke potatoe) it is also fed into your audio seperation node video frames as i still suspect that sets the length... but not sure. 

Ok so second hurdle jumped....sort off.... the workflow below works but the spectrograph looks a bit blocky (its not in the right spot but i see how to change that) and i dont know if there is a way to stop the imageCompositeMasked from using all my ram? I dont actually know if I have made mistakes on this that is causing that.... just because its working now does not mean its working...well lol. Any tips/ advice greatfully recieved. 


[Log create(1).json](https://github.com/user-attachments/files/17851362/Log.create.1.json)
Result: https://www.youtube.com/watch?v=hdghIgUnD2s

I need to up the frame rate but thats a bit of a worry given the ram usage on imageCompositeMasked at mo lol...there has to be a solution to that.
